### PR TITLE
chore: use npm api for "latest" tag

### LINKED ISSUE #292: Fallback to the original npm registry (or jsdelivr) when unpkg fail
For couple of hours I got error 500. First noticed that a ton of badges are just broken on ton of repos. So then I tried to create new project (for which I use mine `charlike` scaffolder which is again using the unpkg registry too) it turns out that unpkg returns 500 for non-cached packages (which is still strange cuz it should have mine pacakges already cached, but probably some cache invalidation happen in recent months).

So, we need to handle this case and fallback to some more reliable source of truth. I bet on jsdelivr, cuz as we know the npm registry sucks a bit.

### PR DESCRIPTION
**Pull request by amio**
prefer official npm api as much as possible

resolve #292

### CODE REVIEW

File: api/npm.ts [hunk 1]
```diff
@@ -66,17 +67,27 @@ async function handler ({ topic, scope, pkg, tag }: PathArgs) {
...
+  const meta = await (tag === 'latest' && pkg[0] !== '@' ? npmMetadata(pkg) : pkgJson(pkg, tag))
 
   switch (topic) {
     case 'version': {
       return {
-        subject: `npm${tag === 'latest' ? '' : '@' + tag}`,
+        subject: tag === 'latest' ? 'npm' : `npm@${tag}`,
         status: version(meta.version),
-        color: tag === 'latest' ? versionColor(meta.version) : 'cyan'
+        color: tag === 'latest' ? versionColor(meta.version) : 'blue'
```
Comment (tunnckoCore) [inline api/npm.ts]: hm, why always blue if not `latest`? it was cool that it's a bit different.

Comment (tunnckoCore) [reply api/npm.ts]: I think that no matter the dist-tag it should follow the `versionColor`. But when it's on some dist-tag AND above v1 then to be cyan.

Comment (amio) [reply api/npm.ts]: Let's go with __no matter the dist-tag it should follow the versionColor__

Comment (amio) [reply api/npm.ts]: Non-latest tag will be shown on badge label:

[](https://badgen.net/npm/v/yarn/legacy)

Comment (amio) [reply api/npm.ts]: Updated in 1999c2a

Comment (tunnckoCore) [reply api/npm.ts]: Good.

But why just decided to switch from the cyan, in the first place? :thinking:

Comment (amio) [reply api/npm.ts]: `cyan` is used for some unstable/future tag (alpha, canary, rc, etc), not suitable for tag like `legacy`, so I changed it to a more general/meaningless `blue` 😋

Comment (tunnckoCore) [reply api/npm.ts]: ooh is there still such detection in the versionColor, cool then.

**Pull request merged**